src/app/register/page.tsx

```diff
@@ -34,51 +86,137 @@ const Registration = () => {
+                'border-medium-gray': !formData.name,
+              },
+            )}
             type="text"
             name="name"
             value={formData.name}
             onChange={handleInputChange}
+            placeholder="Name"
+            required
           />
         </label>
-        <label className="mb-4 w-full">
-          Email:
+        <label className="w-full">
+          <span className="sr-only">Email</span>
           <input
-            className="mt-2 w-full border border-black p-2"
+            className={cx(
+              'mt-8 w-full border px-5 py-3 placeholder-dark-gray focus:outline-none',
+              {
+                'border-primary':
+                  formData.email && !emailError && !errorMessage,
+                'border-red-500': emailError || errorMessage,
+                'border-medium-gray': !emailError && !formData.email,
+              },
+            )}
             type="email"
             name="email"
             value={formData.email}
             onChange={handleInputChange}
+            onBlur={validateEmail}
+            placeholder="Email"
+            required
           />
+          {emailError && (
+            <p className="absolute pt-1 text-xs text-secondary">{emailError}</p>
+          )}
         </label>
-        <label className="mb-4 w-full">
-          Password:
+        <label className="mb-8 w-full">
+          <span className="sr-only">Heslo</span>
           <input
-            className="mt-2 w-full border border-black p-2"
-            type="password"
+            className={cx(
+              'mt-8 w-full border py-3 pl-5 pr-10 placeholder-dark-gray focus:outline-none',
+              {
+                'border-primary':
+                  formData.password && !passwordError && !errorMessage,
+                'border-red-500': passwordError || errorMessage,
+                'border-medium-gray': !passwordError && !formData.password,
+              },
+            )}
+            type={showPassword ? 'text' : 'password'}
             name="password"
             value={formData.password}
             onChange={handleInputChange}
+            onBlur={validatePassword}
+            placeholder="Password"
+            required
           />
+          <button onClick={() => setShowPassword(!showPassword)} type="button">
+            <EyeIcon
+              className={cx(
+                'transfotm absolute h-5 w-5 -translate-x-8 -translate-y-3 fill-light-gray',
+                { 'fill-primary': showPassword },
+              )}
+            />
+          </button>
+          {passwordError && (
+            <p className="absolute pt-1 text-xs text-secondary">
+              {passwordError}
+            </p>
+          )}
         </label>
-        <button className="mt-4 border border-black px-4 py-2" type="submit">
-          Register
-        </button>
+        {!isLoading ? (
+          <button
+            disabled={isDisabled}
```

Comment:
Nope :D. Dont use disabled as form error. 

Source: https://adamsilver.io/blog/the-problem-with-disabled-buttons-and-what-to-do-instead/